test/test_autograd.py

```diff
@@ -4795,52 +4801,29 @@ def fn(a, b):
                     b = torch.ones(2, dtype=dtype, requires_grad=True)
 
+                    err_msg = fn_id_to_inplace_view_err_msg[fn_id]
+
+                    if not inplace or not output_is_a_view:
+                        gradcheck(fn, (a, b), check_batched_grad=False)
 
-                    if fn_id == "two_output" and inplace and output_is_a_view:
-                        with self.assertRaisesRegex(RuntimeError, err_msg_two_outputs):
-                            fn(a, b)
+                    # Was the custom backward called properly
+                    bw_called[0] = 0
+                    ga_nz[0] = True  # For the case where the backward is called
+
+                    if inplace and output_is_a_view:
+                        with self.assertRaisesRegex(RuntimeError, err_msg):
+                            fn(a, b).backward()
                     else:
-                        # Are the computed gradients correct ?
-                        if inplace and output_is_a_view:
-                            with warnings.catch_warnings(record=True) as w:
-                                if fn_id == "view_of_temp":
-                                    # This will be fixed after the deprecation cycle and the warning becomes
-                                    # an error.
-                                    with self.assertRaisesRegex(RuntimeError,
-                                                                "a view of a leaf Variable that requires grad "
-                                                                "is being used in an in-place operation."):
-                                        gradcheck(fn, (a, b), check_batched_grad=False)
-                                else:
-                                    # This works but the custom backward is not called (or called with partial)
-                                    # gradients as tested below
-                                    gradcheck(fn, (a, b), check_batched_grad=False)
-                            self.assertTrue(len(w) > 0)
-                        else:
-                            gradcheck(fn, (a, b), check_batched_grad=False)
-
-                        # Was the custom backward called properly
-                        bw_called[0] = 0
-                        ga_nz[0] = True  # For the case where the backward is called
-                        with warnings.catch_warnings(record=True) as w:
-                            if inplace and output_is_a_view and fn_id != "one_output":
-                                with self.assertRaisesRegex(RuntimeError,
-                                                            "a view of a leaf Variable that requires grad "
-                                                            "is being used in an in-place operation."):
-                                    fn(a, b).backward()
-                            else:
-                                fn(a, b).backward()
+                        fn(a, b).backward()
```

Comment:
shouldn't this be a gradcheck call?

Reply:
We still do the gradcheck call on 4807 when either inplace/output_is_a_view is False. For this second part we show that we only need to run fn to trigger the RuntimerError. It should be a stronger test just to do fn here, since we know gradcheck calls fn for sure.

Reply:
I feel like this and l 4807 should be merged then into:
```python
if inplace and output_is_a_view:
    with self.assertRaisesRegex(RuntimeError, err_msg):
        fn(a, b)
    else:
        gradcheck(fn, (a, b), check_batched_grad=False)
```

As one should fail during the forward and the other should compute the proper gradients right?

Reply:
Not sure it can be combined to only call gradcheck because you'll still need to make sure that the number times backward is invoked is equal to the expected, and gradcheck may call backward multiple times.

Reply:
but I guess you can change the test to check if its at least 1

Reply:
Update again: I think its still going to be tricky combining the two because of the test for whether `ga` is all zeros. We should probably still call backward explicitly, i.e., so that it defaults non-zero grad out. Otherwise, in gradcheck's test_undefined_grad we actually call backward w/ all zeros grad_out, which would cause the test to fail.

Reply:
Ho I forgot about the counter below. it's good then!

I would still remove the call to backward in the branch that should fail. We actually want the error to happen in the forward.